Applications/Spire/Include/Spire/Ui/CurrencyComboBox.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef SPIRE_CURRENCY_COMBO_BOX_HPP
+#define SPIRE_CURRENCY_COMBO_BOX_HPP
+#include <QWidget>
+#include "Nexus/Definitions/Currency.hpp"
+#include "Spire/Spire/Spire.hpp"
+
+namespace Spire {
+
+  //! Represents a combo box for selecting Currencies.
+  class CurrencyComboBox : public QWidget {
+    public:
+
+      //! Signals that a Currency was selected.
+      /*!
+        \param currency The Id of the selected Currency.
+      */
+      using SelectedSignal = Signal<void (Nexus::CurrencyId currency)>;
+
+      //! Constructs a CurrencyComboBox.
+      /*!
+        \param parent The parent widget.
+      */
+      explicit CurrencyComboBox(QWidget* parent = nullptr);
```

Comment:
Take the CurrencyDatabase in the constructor by ```const&```, do not use the default currency database, that is intended for testing purposes.